core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1230,6 +1265,22 @@ public Builder addSnapshot(Snapshot snapshot) {
       snapshotsById.put(snapshot.snapshotId(), snapshot);
       changes.add(new MetadataUpdate.AddSnapshot(snapshot));
 
+      if (rowLineage) {
+        ValidationException.check(
+            snapshot.firstRowId() >= nextRowId,
+            "Cannot add a snapshot whose first-row-id (%s) is less than the metadata `last-used-id` (%s) because this will end up generating duplicate row_ids.",
```

Comment:
minor: I don't think we use backticks anywhere else in error messages, so maybe we could just use `'` instead to get `'last-used-id'`? Same for the other error messages in this PR